syntax = "proto3";
import "google/protobuf/empty.proto";
package proto;
option go_package = ".;proto";

service Order {
  // 购物车
  rpc CarItemList(UserInfo) returns (CartItemListResponse);             // 获取用户的所有购物车信息
  rpc CreateCartItem(CartItemRequest) returns (ShopCartInfoResponse);   // 添加商品到购物车
  rpc UpdateCartItem(CartItemRequest) returns (google.protobuf.Empty);  // 修改购物车条目信息
  rpc DeleteCartItem(CartItemRequest) returns (google.protobuf.Empty);  // 删除购物车条目

  // 订单
  rpc CreateOrder(OrderRequest) returns (OrderInfoResponse);            // 新建订单
  rpc OrderList(OrderFilterRequest) returns (OrderListResponse);        // 订单列表
  rpc OrderDetail(OrderRequest) returns (OrderInfoDetailResponse);      // 订单详情
  rpc UpdateOrderStatus(OrderStatus) returns (google.protobuf.Empty);   // 修改订单的支付状态
}

message UserInfo{
  int32 id = 1;
}

message ShopCartInfoResponse {
  int32 id = 1;
  int32 userId = 2;
  int32 goodsId = 3;
  int32 nums = 4;
  bool checked = 5;
}

message CartItemListResponse{
  int32 total = 1;
  repeated ShopCartInfoResponse data = 2;
}

message  CartItemRequest {
  int32 userId = 1;
  int32 goodsId = 2;
  int32 nums = 3;
  bool checked = 4;
}

message OrderRequest {
  int32 id = 1;
  int32 userId = 2;             // 用户id
  string address = 3;           // 收货地址
  string mobile = 4;            // 用户电话
  string name = 5;
  string post = 6;
}

message OrderInfoResponse {
  int32 id = 1;                 // 订单id
  int32 userId = 2;             // 用户id
  string orderSn = 3;           // 订单号
  string payType = 4;           // 支付方式
  string status = 5;            // 订单状态
  string post = 6;              // 留言
  float total = 7;              // 订单金额
  string address = 8;           // 收货地址
  string name = 9;              // 签收人
  string mobile = 10;           // 联系电话
  string addTime = 11;          // 添加订单时间
}

message OrderListResponse {
  int32 total = 1;
  repeated  OrderInfoResponse data = 2;
}

message OrderFilterRequest {
  int32 userId = 1;
  int32 pages = 2;
  int32 pagePerNums = 3;
}

message OrderItemResponse {
  int32 id = 1;
  int32 orderId = 2;
  int32 goodsId = 3;
  string goodsName = 4;
  string goodsImage = 5;
  float goodsPrice = 6;
  int32 nums = 7;
}

message OrderInfoDetailResponse {
  OrderInfoResponse orderInfo = 1;                                      // 订单详情
  repeated OrderItemResponse data = 2;                                  // 商品详情
}

message OrderStatus {
  string OrderSn = 1;
  string status = 2;
  int64 payTime = 3;
}